decomp/semantics/uds.py

```diff
@@ -1042,45 +1132,150 @@ def from_json(cls, jsonfile: Union[str, TextIO]) -> 'UDSDataset':
 
         VALUE in the above may be anything, but if it is
         dictionary-valued, it is assumed to have the following
-        structure:
+        structure (in the case of a NormalizedUDSAnnotation):
 
         ::
 
-            {SUBSPACE_1: {PROP_1_1: {VER_1_1_1: {'value': VALUE,
-                                                 'confidence': VALUE},
-                                     VER_1_1_2: {'value': VALUE,
-                                                 'confidence': VALUE},
-                                      ...},
+            {SUBSPACE_1: {PROP_1_1: {'value': VALUE,
+                                    'confidence': VALUE},
                          ...},
-             SUBSPACE_2: {PROP_2_1: {VER_2_1_1: {'value': VALUE,
-                                                 'confidence': VALUE},
-                                     VER_2_1_2: {'value': VALUE,
-                                                 'confidence': VALUE},
-                                      ...},
+             SUBSPACE_2: {PROP_2_1: {'value': VALUE,
+                                     'confidence': VALUE},
                          ...},
             }
 
-        Parameters
-        ----------
-        jsonfile
-            (path to) file containing annotations as JSON
+        For details on the structure of RawUDSAnnotation, see the
+        docstring for that class.
         """
+        return super().from_json(jsonfile)
 
-        if jsonfile in cls.CACHE:
-            return cls.CACHE[jsonfile]
+    @overrides
+    def items(self):
+        return super().items()
 
-        ext = splitext(basename(jsonfile))[-1]
+    @overrides
+    def __getitem__(self, k):
+        return super().__getitem__(k)
 
-        if isinstance(jsonfile, str) and ext == '.json':
-            with open(jsonfile) as infile:
-                annotation = json.load(infile)
+class RawUDSDataset(UDSAnnotation):
+    """A raw Universal Decompositional Semantics dataset. Unlike
+       NormalizedUDSDataset, RawUDSDataset may have multiple annotations for
+       a particular attribute. Each annotation is associated with
+       an annotator ID, and different annotators may have annotated
+       different numbers of items.
 
-        elif isinstance(jsonfile, str):
-            annotation = json.loads(jsonfile)
+    Parameters
+    ----------
+    annotation
+        mapping from node ids (for node annotations) or pairs 
+        of node ids separated by %% (for edge annotations) to
+        attribute-value pairs, with values having the structure
+        described above.
+    """
+    @overrides
+    def __init__(self, annotation: Dict[str, Dict[str, Any]]):
+        super().__init__(annotation)
+
+        # Node attributes by annotator
+        self.node_attributes_for_annotator = defaultdict(lambda:
```

Comment:
It might be more readable to just define a nested default dict utility function.

```python
def nested_defaultdict(depth: int) -> Union[dict, defaultdict]:
    """Construct a nested defaultdict
    
    The lowest nesting level is a normal dictionary.
    
    Parameters
    ----------
    depth
        The depth of the nesting   
    """
    if depth < 0:
        raise ValueError('depth must be a nonnegative int')
    
    if not depth:
        return dict
    else:
        return defaultdict(lambda: nested_defaultdict(depth-1))
```